### PR TITLE
[MRG] Fixes to ICA.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3252
closes https://github.com/mne-tools/mne-python/issues/3168

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Do you have time to take care of #3168 while you're at it?

**Comment by jaeilepp** [comment]
Yeah, I can do that.

File: tutorials/plot_artifacts_correction_ica.py [hunk 1]
```diff
@@ -176,20 +176,21 @@
...
+icas = [reference_ica] + icas_from_other_data
 template = (0, eog_inds[0])
 
 ###############################################################################
 # Now we can do the corrmap.
-fig_template, fig_detected = corrmap(
-    icas_from_other_data, template=template, label="blinks", show=True,
-    threshold=.8, ch_type='mag')
+fig_template, fig_detected = corrmap(icas, template=template, label="blinks",
+                                     show=True, threshold=.8, ch_type='mag')
```
Comment (jaeilepp) [inline tutorials/plot_artifacts_correction_ica.py]: The problem was that the reference ica was not passed to the corrmap function. So basically it was using a random component from the other run to find correlations.

Comment (Eric89GXL) [inline tutorials/plot_artifacts_correction_ica.py]: Don't you need to seed the `random_state` as well, or no?

File: mne/preprocessing/ica.py [hunk 2]
```diff
@@ -1963,6 +1964,8 @@ def f(x):
         ica.n_samples_ = ica_misc['n_samples_']
     if 'labels_' in ica_misc:
         ica.labels_ = ica_misc['labels_']
+    if 'method' in ica_misc:
+        ica.method = ica_misc['method']
```
Comment (agramfort) [inline mne/preprocessing/ica.py]: can you update a test that would fail on master?

File: tutorials/plot_artifacts_correction_ica.py (hunk 1 above)
Comment (jaeilepp) [inline tutorials/plot_artifacts_correction_ica.py]: The ``eog_inds`` is now automatically detected so the code should work fine.

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**